refactor(core): rename ngInjectorDef to ɵinj and ngInjectable def to ɵprov

kara opened the pull request:
Injector defs are not considered public API, so the property
that contains them should be prefixed with Angular's marker
for "private" ('ɵ') to discourage apps from relying on def
APIs directly.

This commit adds the prefix and shortens the name from
ngInjectorDef to inj. This is because property names
cannot be minified by Uglify without turning on property
mangling (which most apps have turned off) and are thus
size-sensitive.

---


Injectable defs are not considered public API, so the property
that contains them should be prefixed with Angular's marker
for "private" ('ɵ') to discourage apps from relying on def
APIs directly.

This commit adds the prefix and shortens the name from
ngInjectableDef to "prov" (for "provider", since injector defs
are known as "inj"). This is because property names cannot
be minified by Uglify without turning on property mangling
(which most apps have turned off) and are thus size-sensitive.

kara committed:
refactor(core): rename ngInjectorDef to ɵinj

Injector defs are not considered public API, so the property
that contains them should be prefixed with Angular's marker
for "private" ('ɵ') to discourage apps from relying on def
APIs directly.

This commit adds the prefix and shortens the name from
ngInjectorDef to inj. This is because property names
cannot be minified by Uglify without turning on property
mangling (which most apps have turned off) and are thus
size-sensitive.

josephperrott commented on aio/tools/transforms/angular-api-package/processors/filterMembers.js:
```diff
@@ -13,6 +13,8 @@ module.exports = function filterMembers() {
       const isAllowed = ({name}) => !this.notAllowedPatterns.some(re => re.test(name));
 
       docs.forEach(doc => {
+        // docs.statics.forEach(st => if (st === 'ɵinj') { console.log('S HERE')})
```
Are these comments still needed?

kara replied:
gah, thank you

kara committed:
refactor(core): rename ngInjectableDef to ɵprov

Injectable defs are not considered public API, so the property
that contains them should be prefixed with Angular's marker
for "private" ('ɵ') to discourage apps from relying on def
APIs directly.

This commit adds the prefix and shortens the name from
ngInjectableDef to "prov" (for "provider", since injector defs
are known as "inj"). This is because property names cannot
be minified by Uglify without turning on property mangling
(which most apps have turned off) and are thus size-sensitive.

IgorMinar commented on aio/tools/transforms/angular-api-package/index.js:
```diff
@@ -179,7 +179,7 @@ module.exports =
         .config(function(filterMembers) {
           filterMembers.notAllowedPatterns.push(
-            /^ng[A-Z].*Def$/
+            /^ng[A-Z].*Def$/, /^ɵ/
```
do we have any left over *Defs or can we remove the first pattern?

kara replied:
In this commit, there was still `ngInjectableDef` to update, but it should be removable in the second commit (which does the `ngInjectable` rename). Updated it in the fixup: 

https://github.com/angular/angular/pull/33151/commits/b1e3587c1e6e93d899f83d9aee7b891b58759f99

IgorMinar reviewed (approved):
some unit tests are still failing because of missing test updates? otherwise lgtm

kara commented:
[VE presubmit](https://test.corp.google.com/ui#id=OCL:274866140:BASE:274866191:1571168824216:adf32228)

[Ivy presubmit](http://test/OCL:274866140:BASE:274928193:1571186960229:bbed8d3e)

mary-poppins commented:
You can preview f451395 at https://pr33151-f451395.ngbuilds.io/.
You can preview 38719ff at https://pr33151-38719ff.ngbuilds.io/.
You can preview 78c5f45 at https://pr33151-78c5f45.ngbuilds.io/.
You can preview b1e3587 at https://pr33151-b1e3587.ngbuilds.io/.

mary-poppins commented:
You can preview f451395 at https://pr33151-f451395.ngbuilds.io/.
You can preview 38719ff at https://pr33151-38719ff.ngbuilds.io/.
You can preview 78c5f45 at https://pr33151-78c5f45.ngbuilds.io/.
You can preview b1e3587 at https://pr33151-b1e3587.ngbuilds.io/.

gkalpak commented on aio/tools/transforms/angular-api-package/index.js:
```diff
@@ -179,7 +179,7 @@ module.exports =
         .config(function(filterMembers) {
           filterMembers.notAllowedPatterns.push(
-            /^ng[A-Z].*Def$/
+            /^ɵ/
```
Theoretically there is still `ngBaseDef`, but no occurrence of that in framework's public API (afaict).

kara replied:
`ngBaseDef` should be removed in a few days anyway

mary-poppins commented:
You can preview f091133 at https://pr33151-f091133.ngbuilds.io/.

mary-poppins commented:
You can preview 763c7a6 at https://pr33151-763c7a6.ngbuilds.io/.

kara committed:
fixup! refactor(core): rename ngInjectableDef to ɵprov

kara commented:
merge-assistance: global approval

mary-poppins commented:
You can preview e166d22 at https://pr33151-e166d22.ngbuilds.io/.
